src/Octoshift/Extensions/StringExtensions.cs

```diff
@@ -12,5 +13,7 @@ public static class StringExtensions
         public static bool HasValue(this string s) => !s.IsNullOrWhiteSpace();
 
         public static bool ToBool(this string s) => bool.TryParse(s, out var result) && result;
+
+        public static string Clean(this string s) => s.HasValue() ? Regex.Replace(s, @"[^a-z0-9_.-]+", "-", RegexOptions.Compiled | RegexOptions.CultureInvariant | RegexOptions.IgnoreCase) : string.Empty;
```

Comment:
A few things here:
1. IMO it's better to use a more meaningful method name that is self descriptive. If I see `str.Clean()` somewhere in the code it would be impossible to understand what it does w/o looking at the implementation. A better name could be for example, `ReplaceInvalidCharacters(string replaceWith = "-")` or even `ReplaceInvalidCharactersWithDash()`.
2. The Regex can be simplified to `[^\w.-]`, then the `IgnoreCase` option can also be removed. So the entire Regex replace will look like the following:
```c#
Regex.Replace(s, "[^\w.-]", "-", RegexOptions.Compiled | RegexOptions.CultureInvariant)
```

Reply:
I pushed a commit to make these changes